src/contracts/Accounting.test.ts

```diff
@@ -0,0 +1,36 @@
+import { Eth } from 'web3-eth';
+import { HttpProvider } from 'web3-providers';
+import { Environment } from '../Environment';
+import deployment from '../deployments/mainnet';
+import { Accounting } from './Accounting';
+import BigNumber from 'bignumber.js';
+import { toChecksumAddress } from 'web3-utils';
+
+describe('Accounting', () => {
+  let environment: Environment;
+  let accounting: Accounting;
+
+  beforeAll(() => {
+    // TODO: This should be replaced with a local ganache test environment using proper test fixtures.
+    const client = new Eth(new HttpProvider('https://mainnet.melonport.com'));
+    environment = new Environment(client, deployment);
+    accounting = new Accounting(environment, '0x1b66598123fefb8759340d4ea6e4b070c4fc4315');
+  });
+
+  it('should return the default share price', async () => {
+    const result = await accounting.defaultSharePrice();
+    expect(result).toStrictEqual(new BigNumber('1000000000000000000'));
+  });
+
+  it('should return WETH address as native asset', async () => {
+    const result = await accounting.nativeAsset();
+    expect(toChecksumAddress(result)).toBe(
+      toChecksumAddress(environment.deployment.thirdPartyContracts.tokens[4].address),
```

Comment:
Use ```Token.findToken(environment, 'TOKEN');``` for this.

Reply:
`Token.findToken(environment, 'TOKEN');` doesn't exist, as far as I am aware. Shall I write it?

Reply:
Sorry, it's Token.findDefinition. Inherited from AbstractToken.